sigstore/oidc.py

```diff
@@ -77,22 +92,28 @@ def __init__(self, raw_token: str) -> None:
         # certificate binding and issuance.
         try:
             self._unverified_claims = jwt.decode(
-                self._raw_token, options={"verify_signature": False}
+                raw_token,
+                options={
+                    "verify_signature": False,
+                    "verify_aud": True,
+                    "verify_iat": True,
+                    "verify_exp": True,
+                    "require": ["aud", "iat", "exp", "iss"],
+                },
+                audience=DEFAULT_AUDIENCE,
             )
-        except jwt.InvalidTokenError as exc:
-            raise IdentityError("invalid identity token") from exc
+        except Exception as exc:
```

Comment:
This is unfortunately similar to https://github.com/pypi/warehouse/blob/beaa920c0c805c35066dab7239be9daf84ebb1ba/warehouse/oidc/services.py#L242-L276 -- we can't enforce pyjwt's own exception hygiene, so I figured it'd be best to assume it can raise anything rather than play whac-a-mole over time.

That being said, we can catch the more specific known exceptions and leave an `Exception` fallback, if you'd like 🙂

Reply:
Ah, right, forgot about that. This should be fine then.